tests/vmi_multus_test.go

```diff
@@ -766,5 +769,109 @@ func pingVirtualMachine(vmi *v1.VirtualMachineInstance, ipAddr, prompt string) {
 		&expect.BSnd{S: "echo $?\n"},
 		&expect.BExp{R: "0"},
 	}, 30)
-	Expect(err).ToNot(HaveOccurred())
+	Expect(err).ToNot(HaveOccurred(), "Failed to ping VMI %s within the given timeout", vmi.Name)
+}
+
+// Tests in Multus suite are expecting a Linux bridge to be available on each node, with iptables allowing
+// traffic to go through. This function creates a Daemon Set on the cluster (if not exists yet), this Daemon
+// Set creates a linux bridge and configures the firewall. We use iptables-compat in order to work with
+// both iptables and newer nftables.
+// TODO: Once kubernetes-nmstate is ready, we should use it instead
+func configureNodeNetwork(virtClient kubecli.KubevirtClient) {
+	newBool := func(b bool) *bool {
+		return &b
+	}
+
+	newInt64 := func(i int64) *int64 {
+		return &i
+	}
+
+	// Privileged DaemonSet configuring host networking as needed
+	networkConfigDaemonSet := appsv1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "DaemonSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "network-config",
+			Namespace: "kube-system",
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"name": "network-config"},
+			},
+			Template: k8sv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"name": "network-config"},
+				},
+				Spec: k8sv1.PodSpec{
+					Containers: []k8sv1.Container{
+						{
+							Name: "network-config",
+							// Reuse image which is already installed in the cluster. All we need is chroot.
+							// Local OKD cluster doesn't allow use to pull from the outside.
+							Image: "registry:5000/kubevirt/virt-operator:devel",
+							Command: []string{
+								"sh",
+								"-c",
+								"set -x; chroot /host ip link add br10 type bridge; chroot /host iptables -I FORWARD 1 -i br10 -j ACCEPT; touch /tmp/ready; sleep INF",
+							},
+							SecurityContext: &k8sv1.SecurityContext{
+								Privileged: newBool(true),
+								RunAsUser:  newInt64(0),
+							},
+							ReadinessProbe: &k8sv1.Probe{
+								Handler: k8sv1.Handler{
+									Exec: &k8sv1.ExecAction{
+										Command: []string{"cat", "/tmp/ready"},
+									},
+								},
+							},
+							VolumeMounts: []k8sv1.VolumeMount{
+								k8sv1.VolumeMount{
+									Name:      "host",
+									MountPath: "/host",
+								},
+							},
+						},
+					},
+					Volumes: []k8sv1.Volume{
+						k8sv1.Volume{
+							Name: "host",
+							VolumeSource: k8sv1.VolumeSource{
+								HostPath: &k8sv1.HostPathVolumeSource{
+									Path: "/",
+								},
+							},
+						},
+					},
+					HostNetwork: true,
+				},
+			},
+		},
+	}
+
+	// Helper function returning existing network-config DaemonSet if exists
+	getNetworkConfigDaemonSet := func() *appsv1.DaemonSet {
```

Comment:
Does it really need to be a closure? Can't it be an helper function taking the daemonset name?

Reply:
We are using closures a lot in tests, so I (ab)used that to keep the whole logic in a single function. It can be done outside of it, the question is whether one of those options is better. I personally prefer this keeping this (kind of ugly hack) in a tiny package, waiting to be replaced with nmstate. But if you think keeping it outside the function would be better, I'm ok with changing it.

Reply:
Yes makes sense, also because it is going to be replaced. Let's keep it like this